stix_shifter_modules/elastic_ecs/stix_translation/query_constructor.py

```diff
@@ -156,62 +154,27 @@ def _parse_expression(self, expression, qualifier=None) -> str:
                 expression_01 = "({})".format(expression_01)
-            if isinstance(expression.expr2, CombinedComparisonExpression):
+            if isinstance(expression.expr2, StartStopQualifier):
                 expression_02 = "({})".format(expression_02)
-            query_string = "({} {} {})".format(expression_01, operator, expression_02)
-            if qualifier is not None:
-                self.qualified_queries.append("{} {}".format(query_string, qualifier))
-                return ''
-            else:
-                return "{}".format(query_string)
+            return "({} {} {})".format(expression_01, operator, expression_02)
         elif isinstance(expression, ObservationExpression):
-            return self._parse_expression(expression.comparison_expression, qualifier)
-        elif hasattr(expression, 'qualifier') and hasattr(expression, 'observation_expression'):
-            if isinstance(expression.observation_expression, CombinedObservationExpression):
-                operator = self.comparator_lookup[str(expression.observation_expression.operator)]
-                # qualifier only needs to be passed into the parse expression once since it will be the same for both expressions
-                expression_01 = self._parse_expression(expression.observation_expression.expr1)
-                expression_02 = self._parse_expression(expression.observation_expression.expr2, expression.qualifier)
-
-                if expression_01:
-                    return "{expr1}".format(expr1=expression_01)
-            else:
-                return self._parse_expression(expression.observation_expression.comparison_expression, expression.qualifier)
-        elif isinstance(expression, CombinedObservationExpression):
-            operator = self.comparator_lookup[str(expression.operator)]
-            expression_01 = self._parse_expression(expression.expr1)
-            expression_02 = self._parse_expression(expression.expr2)
-            if expression_01 and expression_02:
-                return "({}) {} ({})".format(expression_01, operator, expression_02)
-            elif expression_01:
-                return "{}".format(expression_01)
-            elif expression_02:
-                return "{}".format(expression_02)
-            else:
-                return ''
+            return self._parse_expression(expression.comparison_expression)
+        elif isinstance(expression, StartStopQualifier):
+            return "{} AND {}".format(self._parse_expression(expression.observation_expression), _format_start_stop_qualifier(expression))
         elif isinstance(expression, Pattern):
             return "{expr}".format(expr=self._parse_expression(expression.expression))
         else:
             raise RuntimeError("Unknown Recursion Case for expression={}, type(expression)={}".format(
                 expression, type(expression)))
 
-    def parse_expression(self, pattern: Pattern):
```

Comment:
Is this considered part of the public API? And if so should it be added back with a deprecation warning?

Reply:
Added back without a warning as it matches the structure of the other modules.